### ISSUE TITLE
Automations - false positive for long-running flow runs

### ISSUE THREAD

**Issue by taljaards** [PrefectHQ/prefect#8217]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

I set up an Automation to cancel any flow that has been running for 35 minutes, and then send an email notification. I get notifications for many flow runs, but when I inspect them, I find that they completed sucessfully in times (much) less than 35 minutes.

### Reproduction

I'm not sure. Will Raphaelson wasn't able to reproduce.
All info is available in the *Additional context* section.

### Versions

<details>
  <summary>`prefect version`</summary>

  ```
  Version:             2.7.9
  API version:         0.8.4
  Python version:      3.10.8
  Git commit:          42b80f18
  Built:               Thu, Jan 19, 2023 4:59 PM
  OS/Arch:             win32/AMD64
  Profile:             staljaardrems2com/eta-operations
  Server type:         cloud
  ```

</details>

### Additional context

All info I can think of:

I have an automation that checks all flows; if they have been running for 35 minutes, cancel them, and send a notification. [Screenshot here](https://i.imgur.com/v1QL1uq.png).

I added a notification action to my automation when Automations was released as GA. I then started getting sporadic emails stating that some flow runs were cancelled. I didn't click on every link, but so far I haven't found one that actually ran more than 35 minutes (as per my trigger), and they all completed successfully.

I sent a message on the `#automations-beta` Slack channel with the idea of asking if there are any known issues. Will Raphaelson checked and couldn't reproduce my issue.

My Agent is running on Windows Server 2019 Datacenter Evaluation version 1809 build 17763.3532, Python 3.10.8, and was one version behind - on 2.7.7. 2.7.9 has since been released. I updated to 2.7.9 and left it running for the rest of my working day. I still get false positive emails.

Account ID: `2a0672be-6e64-4081-bdfd-a4dfded5a802`
Workspace ID: `b70a86f1-659f-408f-add7-9665c6bfa327`
Automation ID: `6bc4204f-4ac0-4708-afe3-5d2cd16715b5`

List of example flow runs received via email alert:
- `c5a39dc7-c0dd-4b7f-a845-a7efcb600463` (duration = 41s)
- `f0230486-940f-4616-9a2b-70040a6eeaf0` (duration = 46s)
- `b4d80ae4-70ba-4e05-9327-bb3ae4eb6452` (duration = 48s)
- `887bc37f-edd0-4ef6-ad17-55f747c393ed` (duration = 1m 21s)

Alerts are sent via email block with ID `e040702c-fb78-4a72-b9a2-8359d6677cd2`.
The flows run via a process infra block with ID `34d5b375-ac1e-429e-adbf-38d41fa75548`.

**Comment by madkinsz** [comment]
@WillRaphaelson can you assign a priority?

**Comment by yaronlevi** [comment]
We are also facing this issue.
For our application, it is very critical and reduces a significant part of the functionality related to "automations".
We would be happy to assist.
Please let us know if you require a demonstration of the bug in real-time.

**Comment by yaronlevi** [comment]
@chrisguidry happy to hear that.
GO P-R-E-F-E-C-T!
Let's win the "orchestration wars".

**Comment by nikhiljain12** [comment]
I am getting a similar false alert, but not sure if it's the same issue or if I should file a new bug.

This alert is configured to trigger if a flow-run remains in `Late or Pending state for over 10 minutes`, but we are seeing this even if the flow `Completed` successfully and `in less than 10 minutes`.

> Prefect flow Late or Pending for over 10 minutes
Flow run build-meta/practical-sloth entered state Completed at 2023-02-15T11:05:10.213268+00:00.
Flow ID: e74c9aaf-0e4d-4503-a21e-38a6b0e5b311
Flow run ID: 71bc88ac-f383-48d1-a616-2aaabd79f319
Flow run URL: https://app.prefect.cloud/account/adee12e2-25d9-428a-9a77-32d8d7ca2ac7/workspace/28562586-0812-4e92-803f-99ce64384cbd/flow-runs/flow-run/71bc88ac-f383-48d1-a616-2aaabd79f319
State message: All states completed.

**Comment by chrisguidry** [comment]
Hello, folks, thanks for the reports!  With your help and examples, we've identified and corrected one significant source of misfiring automations.  Those changes are in Prefect Cloud as of today, and we'll be monitoring the situation over the next week.

One thing to note is that the correction will gradually taper in for flow runs starting at the time of the fix, so it's still possible to get some misfires for flow runs that were scheduled earlier.  We should see this taper off over time.

We're also working on another change to give more predictable ordering to events that happen very close together (like a flow run that goes from Running -> Completed within a very short period of time).  We're still working on these changes, and I'm hoping we'll have something in place late next week or so.

**Comment by taljaards** [comment]
Hi @chrisguidry
This has reappeared again, somewhere between 3 and 8 May IIRC, and is still occurring:
![image](https://user-images.githubusercontent.com/6089256/232675944-7db807b0-d9e7-4b74-b430-f38881dd5312.png)

It's for the same automation of last time.
[Here](https://app.prefect.cloud/account/00b3b373-b911-4eb3-ab16-c2d093ae2658/workspace/b70a86f1-659f-408f-add7-9665c6bfa327/flow-runs/flow-run/8cc7e33a-4fcf-459a-bdfb-ad8e3eca45a6)'s an example flow run, which did not get stuck, and in fact Completed sucessfully in 1m 1s.
The [automation](https://app.prefect.cloud/account/00b3b373-b911-4eb3-ab16-c2d093ae2658/workspace/b70a86f1-659f-408f-add7-9665c6bfa327/automations/automation/6bc4204f-4ac0-4708-afe3-5d2cd16715b5):
![image](https://user-images.githubusercontent.com/6089256/232683012-71a8b423-b181-4fa8-bc9a-4693c51292c0.png)

**Comment by WillRaphaelson** [comment]
Hey @taljaards - thanks for raising this. We've identified some new scaling issues with our trigger evaluation service. I know we've said we fixed this before - the increases in traffic have really pushed its limits, sorry about that. It's my top priority to get this stable and monitored, and we'll be executing on a reliability sprint over the next month to ensure this service works consistently and predictably.
